nrf52-hal-common/src/spim.rs

```diff
@@ -155,25 +156,58 @@ impl<T> Spim<T> where T: SpimExt {
             w.enable().enabled()
         );
 
-        // Set to SPI mode 0
-        spim.config.write(|w|
-            w
-                .order().msb_first()
-                .cpha().leading()
-                .cpol().active_high()
-        );
+        // Configure mode
+        spim.config.write(|w| {
+            // Can't match on `mode` as embedded_hal::spi::Mode only derives PartialEq, not Eq
+            if mode == MODE_0 {
+                w.order().msb_first().cpol().active_high().cpha().leading()
+            } else if mode == MODE_1 {
+                w.order().msb_first().cpol().active_high().cpha().trailing()
+            } else if mode == MODE_2 {
+                w.order().msb_first().cpol().active_low().cpha().leading()
+            } else {
+                w.order().msb_first().cpol().active_low().cpha().trailing()
+            }
+        });
 
         // Configure frequency
-        spim.frequency.write(|w|
-            w.frequency().k500() // 500 kHz
-        );
+        spim.frequency.write(|w| match frequency {
+            Frequency::K125 => {
```

Comment:
I wonder if this huge match is nicer or if values for the enum fields & an `unsafe { bits(enum.field) }` was nicer.

Reply:
This would also match what `Uarte` does, I believe

Reply:
Yup, we re-export the PAC type with a nicer name, it would be good to see that here

Reply:
> we re-export the PAC type

Oh, even better!

Reply:
I have made these changes :)